### PR TITLE
add tls support

### PR DESCRIPTION
**Pull request by mpenate**
### Proposed changes
As some high secured environments should have tls comms even for metrics, this PR address to close that gap and makes serve metrics using https an option.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-prometheus-exporter/blob/master/CONTRIBUTING.md) guide
- [x] I have proven my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have ensured the README is up to date
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch on my own fork

### CODE REVIEW

**Comment by pleshakov** [comment]
Hi @mpenate 

Thanks for the PR! We'll review it and get back to you shortly

File: exporter.go
Comment (pleshakov) [inline exporter.go:250]: This is for for consistency with the existing flags (`nginx.ssl-ca-cert`, `nginx.ssl-client-cert`, ...) 

```suggestion
		"Path to the PEM encoded certificate for the nginx-exporter metrics server (when web.secured-metrics=true). The default value can be overwritten by SSL_SERVER_CERT variable.")
```
Could you also update the README accordingly?

File: exporter.go
Comment (pleshakov) [inline exporter.go:253]: This is for for consistency with the existing flags (nginx.ssl-ca-cert, nginx.ssl-client-cert, ...)

```suggestion
		"Path to the PEM encoded key for the nginx-exporter metrics server (when web.secured-metrics=true). The default value can be overwritten by SSL_SERVER_KEY variable.")
```

Could you also update the README accordingly?

File: exporter.go
Comment (pleshakov) [inline exporter.go:416]: previously, `log.Printf("NGINX Prometheus Exporter has successfully started")` was called after all cli argument validation happened. 

now it is called before the validation for the case of web.secured-metrics. 

Could you possibly print that log line after you checked that the cert and key file exist?

File: exporter.go
Comment (pleshakov) [inline exporter.go:420]: for consistency with the other log messages when an error is printed:
```suggestion
			log.Fatalf("Cert file is not set, not readable or non-existent. Make sure you set -web.ssl-server-cert when starting your exporter with -web.secured-metrics=true: %v", err)
```

File: exporter.go
Comment (pleshakov) [inline exporter.go:424]: for consistency with the other log messages when an error is printed:
```suggestion
			log.Fatalf("Key file is not set, not readable or non-existent. Make sure you set -web.ssl-server-key when starting your exporter with -web.secured-metrics=true: %v", err)
```

**Review by pleshakov — COMMENTED** [review]
Hi  @mpenate

Apologies for the delay. 

The PR looks good! Thanks for updating the docs as well. 

Just a few consistency-related issues. Could you possibly address them?

Additionally, we prefer present tense for commit messages. For example, `Add feature` instead of `Added feature`. Could you also update the commit message/PR title?

File: exporter.go
Comment (mpenate) [reply exporter.go]: Fixed

File: exporter.go
Comment (mpenate) [reply exporter.go]: Fixed as suggested

File: exporter.go
Comment (mpenate) [reply exporter.go]: Done, I left different traces to make it easier to detect in log file how it started (https/http)

File: exporter.go
Comment (mpenate) [reply exporter.go]: fixed

File: exporter.go
Comment (mpenate) [reply exporter.go]: done also @ README, thanks!

**Comment by mpenate** [comment]
> Hi @mpenate
> 
> Apologies for the delay.
> 
> The PR looks good! Thanks for updating the docs as well.
> 
> Just a few consistency-related issues. Could you possibly address them?
> 
> Additionally, we prefer present tense for commit messages. For example, `Add feature` instead of `Added feature`. Could you also update the commit message/PR title?

Resolved and also renamed PR/commit message. Thank you for the review!

**Review by pleshakov — APPROVED** [review]
Hi @mpenate 

Looks good to me! Before we could merge, could you possibly rebase the PR against the latest master?

I'll request one more approval from my coworker, as we have 2 approvals policy.

**Review by lucacome — APPROVED** [review]
LGTM

**Comment by mpenate** [comment]
Thanks for your reviews, rebased from upstream master.

**Review by pleshakov — APPROVED** [review]

**Comment by pleshakov** [comment]
@mpenate thanks!

**Pull request merged**
